pandas/core/frame.py

```diff
@@ -2091,10 +2091,18 @@ def to_feather(self, path) -> None:
     @Substitution(klass="DataFrame")
     @Appender(_shared_docs["to_markdown"])
     def to_markdown(
-        self, buf: Optional[IO[str]] = None, mode: Optional[str] = None, **kwargs
+        self,
+        buf: Optional[IO[str]] = None,
+        mode: Optional[str] = None,
+        index: Optional[bool] = None,
+        **kwargs,
     ) -> Optional[str]:
+        if index is not None and "showindex" in kwargs:
+            raise ValueError("Index and showindex are aliases, please choose one")
```

Comment:
I don't really like this error message.  'showindex' is not a valid keyword argument, please use 'index' to control showing the index.

Reply:
oh, so we completely ditch the old kw?